### ISSUE TITLE
Feature request: equivalent of moveaxis (Python/Numpy) and permute (MATLAB)

### ISSUE THREAD

**Issue by Lotte1990** [opencv/opencv#16799]
References:
https://docs.scipy.org/doc/numpy/reference/generated/numpy.moveaxis.html
https://nl.mathworks.com/help/matlab/ref/permute.html

As far as I know, this operation is not available in OpenCV. About a year ago, someone on Stack Overflow also asked for an easy way to do this, but unfortunately the post remains unanswered.
See https://stackoverflow.com/questions/55117581/how-to-permute-the-axes-on-cvmat-in-c

**Comment by alalek** [comment]
Algorithms which use data moving functions are usually not fully efficient.
So try to exclude moving of huge data arrays first.

Take a look on `cv::dnn::blobFromImage()` to build 4D-tensor from image(s).

---

> permute

Community contribution is welcome (with documentation and tests).

**Comment by dkurt** [comment]
I also think that we do not need any custom configuration for tensors permutation. Usually, `cv::split` and `cv::merge` are enough.

@Lotte1990, maybe you can share which permute is needed in your algorithms? `cv::dnn::blobFromImage()` does `H,W,C -> 1,C,H,W`, in example.

**Comment by Lotte1990** [comment]
I agree that it is mainly useful for small tensors, e.g. 4x4x4x4. When I need to access the 4th dimension it could be convenient to move it to the 1st dimension so tensor.row(x) can be called. Also, it could be useful for testing ideas. A "quick and dirty" implementation so I know that the idea in my head actually works. From there, the code can be optimized further. When it doesn't work, I can drop the idea and move on without writing a full implementation first. For me, this operation would be a useful addition to cv::split and cv::merge.
